File: src/com/timepath/quakec/ide/file/QCFileTypeFactory.java
package com.timepath.quakec.ide.file;

import com.intellij.openapi.fileTypes.FileTypeConsumer;
import com.intellij.openapi.fileTypes.FileTypeFactory;
import org.jetbrains.annotations.NotNull;

/**
 * @author TimePath
 */
public class QCFileTypeFactory extends FileTypeFactory {
    @Override
    public void createFileTypes(@NotNull FileTypeConsumer fileTypeConsumer) {
        fileTypeConsumer.consume(QCFileType.INSTANCE, "qc");
        fileTypeConsumer.consume(QCFileType.INSTANCE, "qh");
        fileTypeConsumer.consume(QCFileType.INSTANCE, "c");
        fileTypeConsumer.consume(QCFileType.INSTANCE, "h");
    }
}

File: src/main/kotlin/com/timepath/quakec/ide/file/QCFileTypeFactory.kt
package com.timepath.quakec.ide.file

import com.intellij.openapi.fileTypes.FileTypeConsumer
import com.intellij.openapi.fileTypes.FileTypeFactory

class QCFileTypeFactory : FileTypeFactory() {
    override fun createFileTypes(consumer: FileTypeConsumer) {
        consumer.consume(QCFileType, "qc")
        consumer.consume(QCFileType, "qh")
    }
}
